src/ucp/tag/eager_rcv.c

```diff
@@ -70,27 +94,33 @@ ucp_eager_handler(void *arg, void *data, size_t length, void *desc,
             /* TODO In case an error status is returned from ucp_tag_process_recv,
              * need to discard the rest of the messages */
             status = UCS_OK;
+
+            if (flags & UCP_RECV_DESC_FLAG_SYNC) {
+                ucp_eager_sync_send_handler(arg, data, flags);
+            }
+
```

Comment:
note that this will re-calculate status because of 'goto out'.
need to remove 'err' label and replace it with 'out'